### PR TITLE
refactor: add no-implicit-coercion rule for ESLint

### PR DESCRIPTION
**Pull request by libook**
## 该 PR 相关 Issue / Involved issue

无

## 完整路由地址 / Example for the proposed route(s)

```routes
/1point3acres/post/hot
/36kr/motif/452
/aliyun/developer/group/alitech
/asahi
/av01/actor/七沢みあ
/av01/tag/中出し
/avgle/videos
/axis-studios/work/full-service-cg-production 
/bangumi/calendar/today
/bangumi/group/boring 
/bangumi/subject/240038
/behance/mishapetrick 
/bilibili/blackboard
/bilibili/user/dynamic/2267573
/bilibili/user/followings/2267573
/bilibili/link/news/live
/bilibili/partion/ranking/171/3
/bilibili/partion/33
/bilibili/weekly
/bjnews/epaper/A 
/blow-studio
/blur-studio
/chinafile/all
/coolapk/dyh/1524
/coolapk/tuwen
/coronavirus/sg-moh
/digic-pictures/works/real-time-engine
/digitaling/articles/latest
/douban/channel/30168934/subject/0
/douban/topic/48823
/douban/explore
/douban/people/62759792/status
/douban/people/exherb/wish/torrentProvider=1337x
/dxy/vaccine/北京
/fanbox/otomeoto
/hhgal
/github/starred_repos/DIYgod
/github/topics/framework
/gov/moa/sjzxfb
/gov/province/jiangsu/important-news
/gov/taiwan/mnd
/gracg/user11968EIcqS3
/guancha
/guancha/personalpage/243983
/hinatazaka46/blog
/hkepc/news
/hket/sran001
/huxiu/collection/212
/ieee/author/37283006000/newest/10
/imaijia/category/xls
/jandan/pic
/japanpost/track/EJ123456789JP/ja
/jike/user/3EE02BC9-C5B3-4209-8750-4ED1EE0F67BB
/jinritoutiao/keyword/AI
/juejin/category/frontend
/juejin/tag/架构
/juejin/trending/ios/monthly
/juejin/books
/juejin/pins/6824710202487472141
/juejin/posts/3051900006845944
/juejin/collections/1697301682482439
/juejin/collection/6845243180586123271
/juejin/shares/56852b2460b2a099cdc1d133
/keyakizaka46/blog
/kongfz/people/5032170
/latepost
/leetcode/submission/us/nathandai
/lfsyd/1
/lizhi/user/27151442948222380
/mastodon/account_id/mastodon.social/23634/statuses/only_media
/mastodon/acct/CatWhitney@mastodon.social/statuses
/matataki/posts/hot
/matataki/posts/latest/ipfs
/matataki/users/9/posts
/matataki/tokens/22/posts/3
/matataki/tags/150/区块链/posts
/matataki/users/3017/favorites/155/posts
/meipai/user/56537299
/method-studios/games
/mi/golden
/miniflux/subscription/categories=科技,影音
/miniflux/feeds=1&2&3/mark=read&limit=7&status=unread
/mp4ba/6
/mp4er
/mpaypass/news
/natgeo/environment/article
/neea/gaokao
/notefolio
/novel/ptwxz/10/10272
/novel/wenxuemi/6/6144
/parcel/hermesuk/[tracking number]
/pcr/news-cn
/pediy/topic/android/digest
/pgyer/kz-test
/pianyuan
/pixiv/user/bookmarks/15288095
/pixiv/ranking/week
/popiask/popi6666
/ps/product/UP9000-CUSA00552_00-THELASTOFUS00000
/sakurazaka46/blog
/showroom/room/93401
/shuhui/comics/1
/siren/news
/sketch/updates
/smzdm/keyword/女装
/soul/posts/hot/NXJiSlM5V21kamJWVlgvZUh1NEExdz09
/soul/Y2w2aTNWQVBLOU09
/ssmh
/ssmh/category/6
/sspai/activity/d0u947vr
/cninfo/announcement/szse/000002/gssz0000002/category_ndbg_szsh
/taptap/changelog/142793
/tencentvideo/playlist/jx7g4sm320sqm7i
/wechat/csm/huxiu_com
/wechat/feeds/MzIwMzAwMzQxNw==
/weixin/miniprogram/devtools
/weixin/miniprogram/framework
/weixin/miniprogram/wxcloud/cloud-sdk
/wechat/mp/homepage/MzA3MDM3NjE5NQ==/16
/wechat/mp/msgalbum/MzA3MDM3NjE5NQ==/1375870284640911361
/wechat-open/community/xyx-question/0
/wechat-open/community/xcx-question/new
/thepaper/839studio/2
/tieba/post/5853240586
/tingshuitz/dalian
/tingshuitz/wuhan
/tuicool/mags/tech
/uisdc/zt/design-history
/unit-image/films/vfx
/bjtu/gs/all
/bupt/yz/int
/hbut/cs/xwdt
/hbut/news/tzgg
/jnu/yw/tt
/nku/jwc/1
/sctu/jwc/13
/sustech/newshub-zh
/szu/yz/1
/ustc/jwc/info
/wtu/2
/zju/career/1
/zju/grs/1
/zju/physics/1
/zzuli/campus/0
/zzuli/yjsc/0
/voa/cantonese/zprtie-ttp 
weatheralarm/广东省
/weibo/timeline/3306934123
/wukong/user/5826687196
/xiaohongshu/board/5db6f79200000000020032df
/xiaohongshu/user/593032945e87e77791e03696/album
/xiaohongshu/user/593032945e87e77791e03696/notes
/xuangubao/subject/41
/yande.re/post/popular_recent
/zfrontier/board/56
/zfrontier/postlist/:byReplyTime
/zhihu/collection/26444956
/zhihu/posts/people/frederchen
/zyshow/chongchongchong
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [ ] New Route
- [ ] Documentation
  - [ ] CN
  - [ ] EN
- [ ] 全文获取 fulltext
  - [ ] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] 日期和时间 date and time
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

### 消除无意义的模板字符串格式

添加ESLint规则no-implicit-coercion，并配置`"disallowTemplateShorthand": true`，可自动检查和修改无意义的模板字符串。

本PR未使用`eslint --fix`，因为自动修改会把所有有问题的模板字符串用`String()`包括起来，多数情况不使用`String()`已经可以得到字符串，所以采用手动改写的方式。

个人脑容量有限，遇到一些可读性不是那么高的代码（比如多层三元运算符嵌套）我进行了一下重构，使得可读性好一点点。

### CODE REVIEW

File: docs/en/joinus/quick-start.md [hunk 1]
```diff
@@ -141,7 +141,7 @@ Create a new js script in [/lib/routes/](https://github.com/DIYgod/RSSHub/tree/m
             list
                 .map((index, item) => {
                     item = $(item);
-                    itemPicUrl = `${item.find('a.cover').attr('style')}`.replace('background-image:url(', '').replace(')', '');
+                    itemPicUrl = String(item.find('a.cover').attr('style')).replace('background-image:url(', '').replace(')', '');
```
Comment (SukkaW) [inline docs/en/joinus/quick-start.md:144]:
```suggestion
                    itemPicUrl = item.find('a.cover').attr('style').replace('background-image:url(', '').replace(')', '');
```

cheerio's `attr` method always returns `string`.

File: docs/joinus/quick-start.md [hunk 2]
```diff
@@ -142,7 +142,7 @@ sidebar: auto
             list
                 .map((index, item) => {
                     item = $(item);
-                    itemPicUrl = `${item.find('a.cover').attr('style')}`.replace('background-image:url(', '').replace(')', '');
+                    itemPicUrl = String(item.find('a.cover').attr('style')).replace('background-image:url(', '').replace(')', '');
```
Comment (SukkaW) [inline docs/joinus/quick-start.md:145]:
```suggestion
                    itemPicUrl = item.find('a.cover').attr('style').replace('background-image:url(', '').replace(')', '');
```

File: lib/routes/av01/tag.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ module.exports = async (ctx) => {
             list
                 .map((index, item) => {
                     item = $(item);
-                    const itemPicUrl = `${item.find('img').attr('data-src')}`;
+                    const itemPicUrl = String(item.find('img')).attr('data-src');
```
Comment (SukkaW) [inline lib/routes/av01/tag.js:27]:
```suggestion
                    const itemPicUrl = item.find('img').attr('data-src');
```

File: lib/routes/gov/moa/sjzxfb.js [hunk 4]
```diff
@@ -11,10 +11,10 @@ module.exports = async (ctx) => {
         title: '中华人民共和国农业农村部 - 数据 - 最新发布',
         link: `http://zdscxx.moa.gov.cn:8080/nyb/pc/index.jsp`,
         item: data.map((item) => ({
-            title: `${item.title}`,
-            description: `${item.content}`,
+            title: item.title,
+            description: item.content,
             link: `http://zdscxx.moa.gov.cn:8080/misportal/public/agricultureMessageViewDC.jsp?page=1&channel=%E6%9C%80%E6%96%B0%E5%8F%91%E5%B8%83&id=${item._id}`,
-            pubDate: `${item.publishTime}`,
+            pubDate: new Date(item.publishTime),
```
Comment (SukkaW) [inline lib/routes/gov/moa/sjzxfb.js:17]: Why change the type from `string` to `Date`?

File: lib/routes/gov/taiwan/mnd.js [hunk 5]
```diff
@@ -48,7 +48,7 @@ module.exports = async (ctx) => {
                 const content = cheerio.load(detailResponse.data);
 
                 item.description = content('div.thisPages').html();
-                item.link = content('form[name="aspnetForm"]').attr('action').replace('.', `${rootUrl}`);
+                item.link = String(content('form[name="aspnetForm"]').attr('action')).replace('.', rootUrl);
```
Comment (SukkaW) [inline lib/routes/gov/taiwan/mnd.js:51]:
```suggestion
                item.link = content('form[name="aspnetForm"]').attr('action').replace('.', rootUrl);
```

File: lib/routes/mp4er/index.js [hunk 6]
```diff
@@ -59,7 +59,14 @@ module.exports = async (ctx) => {
...
 
-                item.description = content('.info0').html() + `<div><b>下载地址：</b>${downloadLinks}</div>` + `${torrents.html() ? `<div><b>种子列表：</b>${torrents.html()}</div>` : ''}`;
+                item.description = `
+                    ${content('.info0').html()}
+                    <div>
+                        <b>下载地址：</b>
+                        ${downloadLinks}
+                    </div>
+                    ${torrents.html() ? `<div><b>种子列表：</b>${torrents.html()}</div>` : ''}
+                `;
```
Comment (SukkaW) [inline lib/routes/mp4er/index.js:69]: This might create unnecessary space in the output.

**Review by SukkaW — CHANGES_REQUESTED** [review]

File: lib/routes/ps/product.js [hunk 7]
```diff
@@ -73,9 +73,9 @@ module.exports = async (ctx) => {
         url: `https://store.playstation.com/valkyrie-api/${langurl}/999/resolve/${gridName}?depth=2`,
     });
     const data = response.data;
-    const item = data.included && data.included.filter((item) => item.id === `${gridName}`);
+    const item = data.included?.filter((item) => item.id === gridName) ?? [];
     ctx.state.data = {
-        title: `PlayStation Store - ${item[0].attributes.name}`,
+        title: `PlayStation Store - ${item?.[0]?.attributes?.name ?? 'Unknown Name'}`,
```
Comment (SukkaW) [inline lib/routes/ps/product.js:78]:
```suggestion
        title: `PlayStation Store - ${item[0].attributes?.name ?? 'Unknown Name'}`,
```

Unnecessary optional chain. `item` will always be an Array.

**Review by SukkaW — CHANGES_REQUESTED** [review]

File: lib/routes/gov/moa/sjzxfb.js (hunk 4 above)
Comment (SettingDust) [reply lib/routes/gov/moa/sjzxfb.js:17]: 苏卡卡,是规范  https://docs.rsshub.app/joinus/pub-date.html

File: lib/routes/mp4er/index.js (hunk 6 above)
Comment (libook) [reply lib/routes/mp4er/index.js:69]: Does this effect the rendered content?
I thought that the source code could be more easy to read.
So what is your suggestion?

Comment (libook) [reply lib/routes/mp4er/index.js]: [`094849a` (#8175)](https://github.com/DIYgod/RSSHub/pull/8175/commits/094849a4552f735964e00b0e345fd1027f303f87)

How about this?

Comment (SukkaW) [reply lib/routes/mp4er/index.js]: LGTM then~

**Review by SukkaW — APPROVED** [review]

**Pull request merged**